api/pkg/apis/v1alpha1/managers/solution/solution-manager_test.go

```diff
@@ -355,8 +355,8 @@ func TestMockGetTwoTargets(t *testing.T) {
 	}
 	state, components, err := manager.Get(context.Background(), deployment, "")
 	assert.Nil(t, err)
-	assert.Equal(t, 2, len(components))
-	assert.Equal(t, 4, len(state.TargetComponent))
+	assert.Equal(t, 0, len(components))
```

Comment:
I am a little confused why this test passed previously? Looks like it should be 0 before reconciling.

Reply:
This is because the provider was initialized with an empty id. And because the cache is shared, leftovers from previous test cases made this case pass.